core/src/main/scala/fs2/Chunk.scala

```diff
@@ -200,20 +288,105 @@ object Chunk {
       (0 until size).foldLeft(z)((z,i) => f(z, at(i)))
     def foldRight[B](z: B)(f: (Boolean,B) => B): B =
       ((size-1) to 0 by -1).foldLeft(z)((tl,hd) => f(at(hd), tl))
+
+    // here be dragons; modify with EXTREME care
+    override def map[B](f: Boolean => B): Chunk[B] = {
```

Comment:
Cool, that's what what I expected. I really wish `Function1` was specialized for every primitive type in both arguments.